spec.bs

```diff
@@ -248,34 +260,49 @@ A user agent holds an <dfn>aggregatable report cache</dfn>, which is a [=list=]
+
+<dfn>Minimum report delay</dfn> is a non-negative [=duration=] that controls the minimum delay to deliver an [=aggregatable report=]
 
-<dfn>Max report contributions</dfn> is a positive integer that controls how many contributions can be present in a single report.
+<dfn>Randomized report delay</dfn> is a positive [=duration=] that controls the random delay to deliver an [=aggregatable report=]. This delay is additional to the minimum delay.
 
 Issue: More
 
 Algorithms {#algorithms}
 ====================
 
-<!-- A {{WorkletGlobalScope}} |scope|'s <dfn algorithm for="PrivateAggregation">api identifier</dfn> is the result of:
-1. If |scope|'s [=worklet global scope type=] is {{FledgeWorkletGlobalScope}}, then return "<code>[=context type/fledge=]</code>".
-1. [=Assert=]: |scope|'s [=worklet global scope type=] is {{SharedStorageWorkletGlobalScope}}.
-1. Return "<code>[=context type/shared-storage=]</code>". -->
+To <dfn>serialize an integer</dfn>, represent it as a [=string=] of the shortest possible decimal number.
+
+Issue: This would ideally be replaced by a more descriptive algorithm in Infra. See infra/201
+
+Scheduling reports {#scheduling-reports}
+----------------------------------------
 
-To <dfn export algorithm for="PrivateAggregation">obtain a reporting endpoint</dfn> given an [=origin=] |reportingOrigin| and [=context type=] |api|, perform the following steps. They return a [=URL=].
+To perform the <dfn algorithm for="PrivateAggregation">report creation and scheduling steps</dfn> with an [=origin=] |reportingOrigin|, a [=context type=] |api| and a [=list=] of {{PAHistogramContribution}}s |contributions|:
 1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
-1. Let |path| be the [=string/concatenation=] of «"<code>.well-known/private-aggregation/report-</code>", |api|».
-1. Let |base| be the result on running the [=URL parser=] on the [=serialization of an origin|serialization=] of |reportingOrigin|.
-1. [=Assert=]: |base| is not failure.
-1. Let |result| be the result of running the [=URL parser=] on |path| with |base|.
-1. [=Assert=]: |result| is not failure.
-1. Return |result|.
+1. Let |truncatedContributions| be a new [=list/is empty|empty=] [=list=].
+1. If |contributions| has a [=list/size=] greater than [=maximum report contributions=]:
+    1. [=set/For each=] |n| of [=the exclusive range|the range=] 0 to [=maximum report contributions=], exclusive:
+        1. [=set/Append=] |contributions|[|n|] to |truncatedContributions|.
+1. Otherwise, set |truncatedContributions| to |contributions|.
+1. Let |contributionSum| be 0.
+1. [=set/iterate|For each=] |contribution| of |truncatedContributions|:
+    1. [=Assert=]: |contribution|["|value|"] is non-negative.
+    1. Add |contribution|["|value|"] to |contributionSum|.
+1. Let |sufficientBudget| be the result of [=PrivateAggregation/consuming budget if permitted=] given |contributionSum|, |reportingOrigin|, |api| and the [=current wall time=].
```

Comment:
nit: Shall we put current wall time in a variable and used later?

Reply:
Makes sense! Done